suprb/optimizer/rule/selection.py

```diff
@@ -27,10 +27,23 @@ class RouletteWheel(RuleSelection):
     """Selection probability is proportional to fitness."""
 
     def __call__(self, rules: list[Rule], random_state: RandomState, size: int = 1) -> list[Rule]:
-        fitness = np.array([rule.fitness_ for rule in rules])
-        fitness /= np.sum(fitness)
-
-        return random_state.choice(rules, p=fitness, size=size)
+        try:
```

Comment:
Isn't it possible to rewrite this? 

```
rules = [rule for rule in rules if rule.fitness_ != -np.inf]
if rules:
    fitnesses = np.array([rule.fitness_ for rule in rules])
    weights = fitnesses / np.sum(fitnesses)

     return random_state.choice(rules, p=weights, size=size)
else:
     return random_state.choice(rules, size=size)
```

Reply:
I originally did it the more complicated way to avoid unneeded computations (as we do this quite a lot). But I agree that your version is much better readable